src/main/java/ee/ut/EventsController.java

```diff
@@ -236,8 +237,24 @@ public void updateItem(Void item, boolean empty) {
             private final Button button = new Button("Kustuta");
             {
                 button.setOnAction(event -> {
-                    deleteEvent(getData(), getTableRow().getItem());
-                    savedEvents.refresh();
+                    //Confirmation on deleting
+                    Alert alert = new Alert(Alert.AlertType.CONFIRMATION);
+                    alert.setTitle("Kinnita kustutamine");
+                    alert.setHeaderText("Kinnita kustutamine");
+                    alert.setContentText("Oled kustutamas sündmust. Kas soovid jätkata?");
+
+                    ButtonType buttonTypeDelete = new ButtonType("Kustuta");
+                    ButtonType buttonTypeCancel = new ButtonType("Katkesta");
+
+                    alert.getButtonTypes().setAll(buttonTypeDelete, buttonTypeCancel);
+
+                    Optional<ButtonType> result = alert.showAndWait();
+                    if (result.get() == buttonTypeDelete) {
+                        deleteEvent(getData(), getTableRow().getItem());
+                        savedEvents.refresh();
+                    } else if (result.get() == buttonTypeCancel) {
```

Comment:
As far as I can see, this else if branch gives no real benefit - there is nothing else trying to catch this actionevent as far as I know.